src/sentry/migrations/0915_dashboard_starred_add_position_column_and_constraints.py

```diff
@@ -0,0 +1,92 @@
+    ):
+        favorite.organization = favorite.dashboard.organization
+        favorites_to_update.append(favorite)
+
+        if len(favorites_to_update) >= batch_size:
+            DashboardFavoriteUser.objects.bulk_update(
+                favorites_to_update, ["organization"], batch_size=batch_size
+            )
+            favorites_to_update = []
+
+    # Update remaining items
+    if favorites_to_update:
+        DashboardFavoriteUser.objects.bulk_update(
+            favorites_to_update, ["organization"], batch_size=batch_size
+        )
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
+
+    dependencies = [
+        ("sentry", "0914_increase_orgmember_user_email_max_length"),
+    ]
+
+    operations = [
+        migrations.AlterUniqueTogether(
+            name="dashboardfavoriteuser",
+            unique_together=set(),
+        ),
+        migrations.AddField(
+            model_name="dashboardfavoriteuser",
+            name="organization",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.organization"
+            ),
+        ),
+        migrations.AddField(
+            model_name="dashboardfavoriteuser",
+            name="position",
+            field=models.PositiveSmallIntegerField(db_default=None, null=True),
+        ),
+        migrations.AddConstraint(
+            model_name="dashboardfavoriteuser",
+            constraint=models.UniqueConstraint(
+                deferrable=django.db.models.constraints.Deferrable["DEFERRED"],
+                fields=("user_id", "dashboard"),
+                name="sentry_dashboardfavoriteuser_unique_favorite_per_dashboard",
+            ),
+        ),
+        migrations.RunPython(
+            code=backfill_dashboard_organization,
+            reverse_code=migrations.RunPython.noop,
+            hints={"tables": ["sentry_dashboardfavoriteuser"]},
+        ),
```

Comment:
Is this going to run quickly enough to complete during a deploy? You may want to separate the schema changes and the backfill operation into two migrations and run the backfill as a post deploy.

Reply:
🤔 2k should be ok.